lake/commits/snapshot.go

```diff
@@ -16,44 +16,47 @@ var (
 )
 
 type View interface {
-	Lookup(ksuid.KSUID) (*segment.Reference, error)
-	Select(extent.Span, order.Which) Segments
-	SelectAll() Segments
+	Lookup(ksuid.KSUID) (*data.Object, error)
+	Select(extent.Span, order.Which) DataObjects
+	SelectAll() DataObjects
 }
 
 type Writeable interface {
 	View
-	AddSegment(seg *segment.Reference) error
-	DeleteSegment(id ksuid.KSUID) error
+	AddDataObject(*data.Object) error
+	DeleteObject(id ksuid.KSUID) error
```

Comment:
Maybe `DeleteDataObject` for symmetry with `AddDataObject`?

Reply:
Right now, DeleteObject only takes an id and works for both index and data objects.  I think this should change so that you can store actions in a pool and make the pool key(s) the span boundaries.  If you don't have the span in the delete you can do range queries to get the related Adds/Deletes.  When we make this change, we can rename DeleteObject and we will have DeleteDataObject and DeleteIndexObject and maybe some other DeleteFooObejcts.